File: Python/leetcode_506.py
from typing import List


def find_relative_ranks(score: List[int]) -> List[str]:
    rank_list = ["Gold Medal", "Silver Medal", "Bronze Medal"]

    # Sort scores in descending order with original indices
    sorted_scores = sorted(enumerate(score), key=lambda x: -x[1])

    result = [""] * len(score)

    for i, (idx, val) in enumerate(sorted_scores):
        if i < 3:
            result[idx] = rank_list[i]
        else:
            result[idx] = str(i + 1)

    return result


print(find_relative_ranks([1]))

File: JavaScript/leetcode_506.js
var findRelativeRanks = function (score) {
    const rankList = ["Gold Medal", "Silver Medal", "Bronze Medal"];

    // Create an array of [index, score] pairs
    const indexedScores = score.map((val, idx) => [idx, val]);

    // Sort descending by score
    indexedScores.sort((a, b) => b[1] - a[1]);

    // Prepare result array
    const result = new Array(score.length);

    // Assign ranks and medals
    for (let i = 0; i < indexedScores.length; i++) {
        const index = indexedScores[i][0];
        if (i < 3) {
            result[index] = rankList[i];
        } else {
            result[index] = (i + 1).toString();
        }
    }

    return result;
};
